R-packages/covidcast/R/covidcast.R

```diff
@@ -256,16 +277,128 @@ covidcast_signal <- function(data_source, signal,
   }
 
   df <- covidcast_days(data_source, signal, start_day, end_day, geo_type,
-                       geo_values, as_of, issues, lag, 
+                       geo_values, as_of, issues, lag,
                        relevant_meta$num_locations)
 
   # Drop direction column (if it still exists)
   df$direction <- NULL
 
-  # Assign covidcast_signal class and add some helpful attributes
-  class(df) <- c("covidcast_signal", "data.frame")
-  attributes(df)$metadata <- relevant_meta
-  return(df)
+  return(as.covidcast_signal(df, signal, geo_type, data_source,
+                             metadata = relevant_meta))
+}
+
+#' Convert data from an external source into a form compatible with
+#' `covidcast_signal`.
+#'
+#' Several methods are provided to convert common objects (such as data frames)
```

Comment:
Yes, this might be unnecessary future-proofing. The identity function exists in case we want to put `as.covidcast_signal()` at the top of various internal functions so they can accept ordinary data frames, but that likely won't work because of the need to specify arguments to `as.covidcast_signal`.

We could add additional methods, although right now the only other thing I can imagine converting would be a [tsibble](https://otexts.com/fpp3/tsibble-objects.html).

I don't think this means we shouldn't use a generic method; base R includes several generics with only one method, like `as.dist` and `as.table`. But we could change the documentation here.

Reply:
I think this is ok to leave as is then.